libsolidity/Types.cpp

```diff
@@ -333,9 +333,10 @@ MemberList const& ContractType::getMembers() const
 	// We need to lazy-initialize it because of recursive references.
 	if (!m_members)
 	{
+		auto interfaceFunctions = m_contract.getInterfaceFunctions();
 		map<string, shared_ptr<Type const>> members;
-		for (FunctionDefinition const* function: m_contract.getInterfaceFunctions())
-			members[function->getName()] = make_shared<FunctionType>(*function, false);
+		for (auto it = interfaceFunctions.cbegin(); it != interfaceFunctions.cend(); ++it)
```

Comment:
Hehe .. you really don't like how iterators look? I can change it in the places that looks readable. But I don't think it's bad as is.